cmd/kueuectl/app/list/list_pods.go

```diff
@@ -0,0 +1,386 @@
+)
+
+const (
+	podLong = `Lists all pods that matches the given criteria: Should be part of the specified Job kind,
+belonging to the specified namespace, matching
+the label selector or the field selector.`
+	podExample = `  # List Pods
+kueuectl list pods --for job/job-name`
+)
+
+var jobControllersWithPodLabelSelector = []jobControllerWithPodLabelSelector{
+	&kueuejob.Job{},
+	&kueuejobset.JobSet{},
+	&kueuemxjob.JobControl{},
+	&kueuepaddlejob.JobControl{},
+	&kueuetfjob.JobControl{},
+	&kueuepytorchjob.JobControl{},
+	&kueuexgboostjob.JobControl{},
+	&kueuempijob.MPIJob{},
+	&pod.Pod{},
+	&kueueraycluster.RayCluster{},
+	&kueuerayjob.RayJob{},
+}
+
+type PodOptions struct {
+	PrintFlags *genericclioptions.PrintFlags
+
+	Limit                  int64
+	AllNamespaces          bool
+	ServerPrint            bool
+	Namespace              string
+	LabelSelector          string
+	FieldSelector          string
+	UserSpecifiedForObject string
+	ForName                string
+	ForGVK                 schema.GroupVersionKind
+	ForObject              *unstructured.Unstructured
+	PodLabelSelector       string
+
+	Clientset k8s.Interface
+
+	genericiooptions.IOStreams
+}
+
+type jobControllerWithPodLabelSelector interface {
+	Object() client.Object
+	GVK() schema.GroupVersionKind
+	PodLabelSelector() string
+}
+
+func NewPodOptions(streams genericiooptions.IOStreams) *PodOptions {
+	return &PodOptions{
+		PrintFlags: genericclioptions.NewPrintFlags("").WithTypeSetter(scheme.Scheme),
+		IOStreams:  streams,
+	}
+}
+
+func NewPodCmd(clientGetter util.ClientGetter, streams genericiooptions.IOStreams) *cobra.Command {
+	o := NewPodOptions(streams)
+
+	cmd := &cobra.Command{
+		Use:                   "pods --for TYPE[.API-GROUP]/NAME",
+		DisableFlagsInUseLine: true,
+		Aliases:               []string{"po"},
+		Short:                 "List Pods belong to a Job Kind",
+		Long:                  podLong,
+		Example:               podExample,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cmd.SilenceUsage = true
+			err := o.Complete(cmd, clientGetter)
+			if err != nil {
+				return err
+			}
+			if o.ForObject == nil {
+				return nil
+			}
+			return o.Run(cmd.Context(), clientGetter)
+		},
+	}
+
+	o.PrintFlags.AddFlags(cmd)
+
+	addAllNamespacesFlagVar(cmd, &o.AllNamespaces)
+	addFieldSelectorFlagVar(cmd, &o.FieldSelector)
+	addLabelSelectorFlagVar(cmd, &o.LabelSelector)
+	addForObjectFlagVar(cmd, &o.UserSpecifiedForObject)
+
+	_ = cmd.MarkFlagRequired("for")
+
+	return cmd
+}
+
+// Complete takes the command arguments and infers any remaining options.
+func (o *PodOptions) Complete(cmd *cobra.Command, clientGetter util.ClientGetter) error {
+	var err error
+
+	o.Limit, err = listRequestLimit()
+	if err != nil {
+		return err
+	}
+
+	outputOption := cmd.Flags().Lookup("output").Value.String()
+	if strings.Contains(outputOption, "custom-columns") || outputOption == "yaml" || strings.Contains(outputOption, "json") {
+		o.ServerPrint = false
+	}
```

Comment:
We don't have `custom-columns`. Maybe like  this? 
```suggestion
		outputOption := ptr.Deref(o.PrintFlags.OutputFormat, "")
	if outputOption == "" || strings.Contains(outputOption, "wide") {
		o.ServerPrint = true
	}
```

Reply:
That make sense. Thanks!